src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1243,8 +1245,18 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                             break;
                         }
 
+                    case DeclarationModifiers.File:
+                        if (!IsFeatureEnabled(MessageID.IDS_FeatureFileTypes) && !ShouldContextualKeywordBeTreatedAsModifier(parsingStatementNotDeclaration: false))
```

Comment:
i don't love hte feature flag check actually controlling behavior.  i think we should just parse this out if ShouldContextualKeywordBeTreatedAsModifier returns true and then report an issue later if the feature is not available.  right?

Reply:
That's fair. This is also the cause of a breaking change in the new language version. This is also what we are doing for 'required' modifier. See [related discussion](https://github.com/dotnet/roslyn/pull/58431/files?diff=unified&w=1#r780601722).